Rewatch.lua

```diff
@@ -307,22 +312,38 @@ function rewatch_SetTooltip(data)
 	end;
 end;
 
+-- gets the name of a spell
+-- spellId: the spell ID of the spell to get its name
+-- return: the corresponding spellName 
+function rewatch_GetSpellName(spellId)
+	local spell = Spell:CreateFromSpellID(spellId);
+	local spellName = spell:GetSpellName();
+	return spellName;
+end
+
 -- gets the spell ID of the highest rank of the specified spell
 -- spellName: the name of the spell to get the highest ranked spellId from
 -- return: the corresponding spellId
 function rewatch_GetSpellId(spellName)
 	-- get spell info and highest rank, return if the user can't cast it (not learned, etc)
-	local name, rank, icon = GetSpellInfo(spellName);
+	local name, rank, icon, _, _, _, spellId = GetSpellInfo(spellName);
```

Comment:
My previous reviewer for BFA did not notice this is broken; are you testing this on a beta WoW version or something?

Reply:
This is not broken per se, but it is required for the changes at lines 338 ff.

I refer to my comment, mistakenly, added at line 352.

Reply:
Sorry, I fail to understand what this change would actually mean. I mean, I can see what it does, but I do not understand if or how it is better. Probably this increases performance, since you only loop the ranks of that one spell instead of the full spell book, but I cannot see the 'cost' of the `Spell:CreateFromSpellID` and `spell:ContinueOnSpellLoad`.

I will follow your judgement here; please tell me if you think this should be copied to the 61000 release version, and I will! 😁

Reply:
I'm not sure if there are any performance improvements with these changes, but I believe `spell:ContinueOnSpellLoad` is necessary to ensure that certain spell information is returned (rather than nil or empty) after the Blizzard 'Load on demand' backend  changes.

On that basis, I think these changes need to be included in the 61000 version.